src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -148,5 +150,90 @@ public override bool TryMapToProjectedDocumentPosition(RazorCodeDocument codeDoc
             projectedIndex = default;
             return false;
         }
+
+        public override RazorLanguageKind GetLanguageKind(RazorCodeDocument codeDocument, int originalIndex)
```

Comment:
Brought this here from `RazorLanguageEndpoint` and exposed this as a service.

Reply:
@TanayParikh this will probably impact your changes to this pretty heavily. I'd expect conflicts 😄